lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -201,13 +201,19 @@ class IngressRequirerUnitData(DatabagModel):
     """Ingress requirer unit databag model."""
 
     host: str = Field(description="Hostname the unit wishes to be exposed.")
+    prefix: str = Field(description="The prefix to be used for the routing configuration.")
 
     @validator("host", pre=True)
     def validate_host(cls, host):  # noqa: N805  # pydantic wants 'cls' as first arg
         """Validate host."""
         assert isinstance(host, str), type(host)
         return host
 
+    @validator("prefix", pre=True)
+    def validate_prefix(cls, prefix):  # noqa: N805  # pydantic wants 'cls' as first arg
+        """Validate host."""
+        assert isinstance(prefix, str), type(prefix)
```

Comment:
don't think we actually need this. pydantic will cast to string or None if we decide to make it Optional.
should we validate that, if string is nonempty, it has a trailing slash? or will traefik fill that in?